geth/common/profiling.go

```diff
@@ -0,0 +1,53 @@
+package common
+
+import (
+	"os"
+	"path/filepath"
+	"runtime/pprof"
+)
+
+const (
+	CpuProfilingFilename = "status_cpu.prof"
+	MemProfilingFilename = "status_mem.prof"
+)
+
+var (
+	cpuFile *os.File
+	memFile *os.File
+)
+
+// Enables CPU profiling for the current process.
+// While profiling, the profile will be buffered and written to file in folder dataDir
+func StartCPUProfile(dataDir string) error {
+	if cpuFile == nil {
+		var err error
+		cpuFile, err = os.Create(filepath.Join(dataDir, CpuProfilingFilename))
+		if err != nil {
+			return err
+		}
+	}
+
+	return pprof.StartCPUProfile(cpuFile)
```

Comment:
If `StartCPUProfile()` is called several times it will return error.
`return fmt.Errorf("cpu profiling already in use")` which is in source code of `pprof.go`